src/pages/accounts/ForgotPassword.tsx

```diff
@@ -82,36 +129,160 @@ function ForgotPasswordPage() {
 	}, [email]);
 
 	const handleSubmit = async () => {
-		if (!validEmail(email)) {
-			setEmailErr(true);
+		// EMAIL
+		if (validEmail(email.value)) {
+			setEmail({ error: false });
+			setCodeStep(true); // Send validation code
+		} else {
+			email.errorMessage = EMAIL_ERROR_MESSAGE;
+			setEmail({ error: true });
+			return;
+		}
+
+		if (codeStep && validCode(validationCode.value)) {
+			setEmail({ valueSet: true });
+			setValidationCode({ error: false, valueSet: true });
+			setPassStep(true);
+		} else {
+			validationCode.errorMessage = VALIDATION_CODE_ERROR_MESSAGE;
+			if (validationCode.value !== '') {
+				setValidationCode({ error: true });
+			}
+			return;
+		}
+
+		if (passStep && validPass(password.value)) {
+			setPassword({ error: false });
+		} else {
+			password.errorMessage = PASS_ERROR_MESSAGE;
+			if (password.value !== '') {
+				setPassword({ error: true });
+			}
+			return;
+		}
+
+		if (passStep && password.value === confirmPass.value) {
+			setConfirmPass({ error: false });
+			console.log('Complete!');
+		} else {
+			confirmPass.errorMessage = CONFIRM_PASS_ERROR_MESSAGE;
+			if (confirmPass.value !== '') {
+				setConfirmPass({ error: true });
+			}
 			return;
 		}
```

Comment:
I think there is quite a bit of logic in this, I would debatably just create the different logic for the different cases in different functions and do something like this:

```ts
const handleSubmit = async () => {
    if(!codeStep && !passStep) {
        return handleSubmitEmail();
    } else if(codeStep && !passStep) {
        return handleCodeStep();
    } else if(passStep) {
        return handlePassStep();
    }
}
```

Then obviously splitting out the logic to the different functions. It will help make it really clear the 3 steps I think.